compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java

```diff
@@ -721,7 +722,9 @@ public static void checkExecutePermission(Path path) {
 
     private static boolean validateDotSeparatedIdentifiers(String identifiers) {
         Matcher m = separatedIdentifierPattern.matcher(identifiers);
-        return m.matches();
+        Matcher mm = onlyDotsPattern.matcher(identifiers);
```

Comment:
Have we verified whether this change has no impact on other places where this util is being used?

Reply:
Other places where this is used validates either the package names or the module names. So this change will prevent `.` from being used as a package name or a module name.